src/internal_rep.rs

```diff
@@ -2320,11 +2320,36 @@ impl ValidatedCall {
             )));
         }
 
-        // Each argument must match the type the function signature expects
-        let mut args = match &call.class_name {
-            Some(c) => vec![convert_class_name_if_this(c, parent_type)?.get_cil_name()],
-            None => Vec::new(),
-        };
+        let mut args =
+            match &call.parent_name {
+                Some(parent_name) => {
+                    match function_info.class {
```

Comment:
This block is pretty hard to follow with all the indentation.  It looks like this whole match block could get pulled out into its own function (returning `Result<(), CascadeErrors>`), which would help readability a lot